### PR TITLE
Feature/contributors

### PR DESCRIPTION
**Pull request by jrobinAV**
Here is the current result:

image

- The list of Avaiga members is automatically built from github Avaiga organization. It only gets the accounts that accepted to set their Avaiga membership to public.
- The list of contributors is built from the join of all contributors.txt of all public repositories. We filter out the Avaiga members.
- Both list are randomly sorted.

Thx

### CODE REVIEW

File: mkdocs.yml_template [hunk 1]
```diff
@@ -109,7 +109,6 @@ nav:
     - 'Contributing to Taipy': 'contributing/contributing.md'
     - 'Code of conduct': 'contributing/code_of_conduct.md'
   - 'Credits':
-    - 'Development lead': 'credits/development_lead.md'
```
Comment (FabienLelaquais) [inline mkdocs.yml_template:112]: Why keep the 'Credits' title?
And not move this page under the 'Contributing' section. May be we can rename this...

File: tools/_setup_generation/setup.py [hunk 2]
```diff
@@ -101,19 +102,25 @@ def exit(self, setup: Setup):
...
 from .step_getting_started import GettingStartedStep
-
-def run_setup(root_dir: str, steps: List[SetupStep] = [
-    VisElementsStep(),
-    RefManStep(),
-    RestRefManStep(),
-    GuiExtRefManStep(),
-    GettingStartedStep()
-    ]):
+from .step_contributor import ContributorStep
```
Comment (FabienLelaquais) [inline tools/_setup_generation/setup.py]: I would have called this step 'contributors' unless you plan for no more than one.
(and ContributorsStep and so on)

File: tools/_setup_generation/step_contributor.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+
+    def enter(self, setup: Setup):
+        self.PATH = os.path.join(setup.docs_dir, "credits", "contributors.md")
+
+    def get_id(self) -> str:
+        return "contributors"
+
+    def get_description(self) -> str:
+        return "Generating the contributors"
```
Comment (FabienLelaquais) [inline tools/_setup_generation/step_contributor.py]: the contributors list

(you're not generating contributors, are you?)

File: tools/_setup_generation/step_contributor.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    def setup(self, setup: Setup) -> None:
+        self.get_repo_urls()
+        self.get_avaiga_members()
+        self.get_contributors()
+        self.build_content((self.MEMBERS, "[AVAIGA_TEAM_MEMBERS]"), (self.CONTRIBUTORS, "[TAIPY_CONTRIBUTORS]"))
+
+    def get_repo_urls(self):
+        repos = requests.get(self.REPOS).json()
```
Comment (FabienLelaquais) [inline tools/_setup_generation/step_contributor.py]: Are you certain that this works for anonymous users?
I mean no access key, no nothing... and still works?

**Review by FabienLelaquais — APPROVED** [review]
It's a superb first proposal.
Love it.

A few minor remarks though.

File: mkdocs.yml_template (hunk 1 above)
Comment (jrobinAV) [reply mkdocs.yml_template:112]: Yep I agree.

I let's move "Contributors" page to the 2nd page of the "Contributing" doc set (after "Contributing to Taipy" but before "Code of conduct").

File: tools/_setup_generation/setup.py (hunk 2 above)
Comment (jrobinAV) [reply tools/_setup_generation/setup.py]: Yep!

File: tools/_setup_generation/step_contributor.py (hunk 3 above)
Comment (jrobinAV) [reply tools/_setup_generation/step_contributor.py]: I would love to...

File: tools/_setup_generation/step_contributor.py (hunk 4 above)
Comment (jrobinAV) [reply tools/_setup_generation/step_contributor.py]: I think so. I only get public information that you can get on GitHub directly.

**Pull request merged**